av1an-core/src/grain.rs

```diff
@@ -42,19 +85,27 @@ impl TransferFunction {
   #[allow(clippy::wrong_self_convention)]
   pub fn from_linear(self, x: f32) -> f32 {
     match self {
-      TransferFunction::BT1886 => x.powf(1. / 2.8),
+      TransferFunction::BT1886 => {
+        // Scale to a raw cd/m^2 value
+        let luma = x * BT1886_WHITEPOINT;
+
+        // The inverse of the `to_linear` formula:
+        // `(L / α)^(1 / λ) - β = x`
+        (luma / bt1886_alpha()).powf(1.0 / BT1886_GAMMA) - bt1886_beta()
+      }
       TransferFunction::SMPTE2084 => {
+        if x == 0.0 {
+          return 0.0;
+        }
         let linear_pow_m1 = x.powf(PQ_M1);
         (PQ_C2.mul_add(linear_pow_m1, PQ_C1) / PQ_C3.mul_add(linear_pow_m1, 1.)).powf(PQ_M2)
       }
     }
   }
 
+  #[inline(always)]
   pub fn mid_tone(self) -> f32 {
-    match self {
-      TransferFunction::BT1886 => 0.18,
-      TransferFunction::SMPTE2084 => 26. / 10000.,
-    }
+    self.to_linear(0.5)
```

Comment:
This seems like it should call `from_linear` (since `from_linear` takes a value between 0 and 1), but I'm not sure. Could you explain why `to_linear` is called here?

Reply:
The midtone we're seeking here is the perceptual gray midtone according to the transfer function, i.e. based on this transfer function, what value with the transfer function applied appears to be "half brightness". This is calculated by doing the transfer calculation on a mathematically half luma value, i.e. `to_linear(0.5)`, and for e.g. a flat gamma curve of 2.4, results in a midtone of about `0.189`.